test/standalone/childprocess_extrapipe/parent.zig

```diff
@@ -0,0 +1,109 @@
+const builtin = @import("builtin");
+const std = @import("std");
+const ChildProcess = std.ChildProcess;
+const math = std.math;
+const windows = std.os.windows;
+const os = std.os;
+const testing = std.testing;
+const child_process = std.child_process;
+
+const windowsPtrDigits: usize = std.math.log10(math.maxInt(usize));
+const otherPtrDigits: usize = std.math.log10(math.maxInt(u32)) + 1; // +1 for sign
+const handleCharSize = if (builtin.target.os.tag == .windows) windowsPtrDigits else otherPtrDigits;
+
+/// assert: buf can store the handle
+fn handleToString(handle: os.fd_t, buf: []u8) std.fmt.BufPrintError![]u8 {
+    var s_handle: []u8 = undefined;
+    const handle_int =
+        // handle is *anyopaque or an integer on unix-likes Kernels.
+        if (builtin.target.os.tag == .windows) @ptrToInt(handle) else handle;
+    s_handle = try std.fmt.bufPrint(buf[0..], "{d}", .{handle_int});
+    return s_handle;
+}
+
+pub fn main() !void {
+    var gpa_state = std.heap.GeneralPurposeAllocator(.{}){};
+    defer if (gpa_state.deinit()) @panic("found memory leaks");
+    const gpa = gpa_state.allocator();
+
+    var it = try std.process.argsWithAllocator(gpa);
+    defer it.deinit();
+    _ = it.next() orelse unreachable; // skip binary name
+    const child_path = it.next() orelse unreachable;
+
+    // use posix convention: 0 read, 1 write
+    var pipe: if (builtin.os.tag == .windows) [2]?windows.HANDLE else [2]os.fd_t = undefined;
+    if (builtin.os.tag == .windows) {
+        const saAttr = windows.SECURITY_ATTRIBUTES{
+            .nLength = @sizeOf(windows.SECURITY_ATTRIBUTES),
+            .bInheritHandle = windows.TRUE,
+            .lpSecurityDescriptor = null,
+        };
+        try child_process.windowsMakeAsyncPipe(&pipe[0], &pipe[1], &saAttr, .parent_to_child);
+    } else {
+        pipe = try os.pipe(); // leaks on default
+    }
+
+    // write pipe to string + add to command
+    var buf: [handleCharSize]u8 = comptime [_]u8{0} ** handleCharSize;
+
+    const s_handle =
+        if (builtin.os.tag == .windows)
+        try handleToString(pipe[0].?, &buf) // read side of pipe
+    else
+        try handleToString(pipe[0], &buf);
+
+    var child_proc = ChildProcess.init(
+        &.{ child_path, s_handle },
+        gpa,
+    );
+    {
+        // close read side of pipe
+        if (comptime builtin.target.isDarwin()) {
```

Comment:
Hey @kubkon,
I'd like to have a quick feedback before removing the clunkyness by pipe numbers and the used optionals of Windows.

This block feels very meh, especially the `errdefer` statement, which is a footgun. However, I dont know any other way to keep the defaults of "simple init and spawn" and not force the handling to be always done by the user.
This code is more of an example how user-handling of posix_spawn together with keeping the default would looks like and not strictly necessary.

Any opinions what I should do with the posix_spawn stuff, @kubkon ?

Reply:
I think a temporary to keep track in the spawn() method, if the user has set `actions` and `attributes` and not doing `deinit()` sounds more reasonable than "partial deinit on error case".